spec/app/github_repo_fetcher_spec.rb

```diff
@@ -1,17 +1,22 @@
 RSpec.describe GitHubRepoFetcher do
   before :each do
-    CACHE.clear
     stub_request(:get, "https://api.github.com/users/alphagov/repos?per_page=100")
       .to_return(
         body: "[ { \"name\": \"some-repo\", \"default_branch\": \"master\" } ]",
         headers: { content_type: "application/json" },
       )
   end
 
+  let(:private_repo) { double("Private repo", private_repo?: true) }
+
+  def public_repo
+    double("Public repo", private_repo?: false, default_branch: "master")
+  end
```

Comment:
I'm not really following the explanation in 5acf5adba24aa89878c66305c7e773236383077d, aren't `let`s reinitialised for every test which uses them?

Reply:
I've just tried switching it back to a `let`, and it worked! I found it did work when running `github_repo_fetcher_spec.rb` directly, but would fail when running the entire test suite. However, I've just tried a couple of runs and it seems to pass every time, so some wires must have got crossed somewhere.

I've now fixed up the commit to use a `let` instead.

Reply:
Hmmm I think it just happens intermittently. An unrelated change just failed:

```
Failures:

  1) GitHubRepoFetcher#repo fetches a repo from cache if it exists
     Failure/Error: all_alphagov_repos.find { |repo| repo.name == repo_name } || raise("alphagov/#{repo_name} not found")
       #<Double "Public repo"> received unexpected message :name with (no args)
     # ./app/github_repo_fetcher.rb:8:in `block in repo'
     # ./app/github_repo_fetcher.rb:8:in `each'
     # ./app/github_repo_fetcher.rb:8:in `find'
     # ./app/github_repo_fetcher.rb:8:in `repo'
     # ./spec/app/github_repo_fetcher_spec.rb:19:in `block (3 levels) in <top (required)>'

  2) GitHubRepoFetcher#repo raises error if no repo is found
     Failure/Error:
       expect {
         GitHubRepoFetcher.instance.repo("something-not-here")
       }.to raise_error(StandardError)

       expected StandardError, got #<RSpec::Mocks::ExpiredTestDoubleError: #<Double "Public repo"> was originally created in one example...only last for one example, and you need to create a new one in each example you wish to use it for.> with backtrace:
         # ./app/github_repo_fetcher.rb:8:in `block in repo'
         # ./app/github_repo_fetcher.rb:8:in `each'
         # ./app/github_repo_fetcher.rb:8:in `find'
         # ./app/github_repo_fetcher.rb:8:in `repo'
         # ./spec/app/github_repo_fetcher_spec.rb:34:in `block (4 levels) in <top (required)>'
         # ./spec/app/github_repo_fetcher_spec.rb:33:in `block (3 levels) in <top (required)>'
     # ./spec/app/github_repo_fetcher_spec.rb:33:in `block (3 levels) in <top (required)>'

  3) GitHubRepoFetcher#repo fetches a repo from GitHub if it doesn't exist in the cache
     Failure/Error: all_alphagov_repos.find { |repo| repo.name == repo_name } || raise("alphagov/#{repo_name} not found")
       #<Double "Public repo"> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for.
     # ./app/github_repo_fetcher.rb:8:in `block in repo'
     # ./app/github_repo_fetcher.rb:8:in `each'
     # ./app/github_repo_fetcher.rb:8:in `find'
     # ./app/github_repo_fetcher.rb:8:in `repo'
     # ./spec/app/github_repo_fetcher_spec.rb:27:in `block (3 levels) in <top (required)>'

  4) GitHubRepoFetcher#readme retrieves the README content from the GitHub CDN
     Failure/Error: all_alphagov_repos.find { |repo| repo.name == repo_name } || raise("alphagov/#{repo_name} not found")
       #<Double "Public repo"> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for.
     # ./app/github_repo_fetcher.rb:8:in `block in repo'
     # ./app/github_repo_fetcher.rb:8:in `each'
     # ./app/github_repo_fetcher.rb:8:in `find'
     # ./app/github_repo_fetcher.rb:8:in `repo'
     # ./app/github_repo_fetcher.rb:14:in `readme'
     # ./spec/app/github_repo_fetcher_spec.rb:70:in `block (3 levels) in <top (required)>'

  5) GitHubRepoFetcher#readme returns nil if no README exists
     Failure/Error: all_alphagov_repos.find { |repo| repo.name == repo_name } || raise("alphagov/#{repo_name} not found")
       #<Double "Public repo"> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for.
     # ./app/github_repo_fetcher.rb:8:in `block in repo'
     # ./app/github_repo_fetcher.rb:8:in `each'
     # ./app/github_repo_fetcher.rb:8:in `find'
     # ./app/github_repo_fetcher.rb:8:in `repo'
     # ./app/github_repo_fetcher.rb:14:in `readme'
     # ./spec/app/github_repo_fetcher_spec.rb:90:in `block (3 levels) in <top (required)>'

Finished in 1.06 seconds (files took 2.33 seconds to load)
1488 examples, 5 failures
```

Reply:
Ah, that's unfortunate.  Well I'm happy to approve a PR restoring the change.

Reply:
Sadly I just tried it and the issue still persists: https://github.com/alphagov/govuk-developer-docs/runs/1733868114

I'll have to rearchitect the tests I think.